src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -259,206 +151,85 @@ public static void main(String... args) throws IOException {
-            if (cmdLine.hasOption(OPTION_S_NAME) || cmdLine.hasOption(OPTION_C_NAME)
-                    || cmdLine.hasOption(OPTION_P_NAME) || cmdLine.hasOption(OPTION_F_NAME)
-                    || cmdLine.hasOption(OPTION_O_NAME)) {
-                result.add("Option '-t' cannot be used with other options.");
-            }
-            else if (filesToProcess.size() > 1) {
-                result.add("Printing AST is allowed for only one file.");
-            }
-        }
-        else if (cmdLine.hasOption(OPTION_S_NAME)) {
-            if (cmdLine.hasOption(OPTION_C_NAME) || cmdLine.hasOption(OPTION_P_NAME)
-                    || cmdLine.hasOption(OPTION_F_NAME) || cmdLine.hasOption(OPTION_O_NAME)) {
-                result.add("Option '-s' cannot be used with other options.");
-            }
-            else if (filesToProcess.size() > 1) {
-                result.add("Printing xpath suppressions is allowed for only one file.");
-            }
-        }
-        // ensure a configuration file is specified
-        else if (cmdLine.hasOption(OPTION_C_NAME)) {
-            final String configLocation = cmdLine.getOptionValue(OPTION_C_NAME);
-            try {
-                // test location only
-                CommonUtil.getUriByFilename(configLocation);
-            }
-            catch (CheckstyleException ignored) {
-                result.add(String.format("Could not find config XML file '%s'.", configLocation));
-            }
+        final int exitStatus;
 
-            // validate optional parameters
-            if (cmdLine.hasOption(OPTION_F_NAME)) {
-                final String format = cmdLine.getOptionValue(OPTION_F_NAME);
-                if (!PLAIN_FORMAT_NAME.equals(format) && !XML_FORMAT_NAME.equals(format)) {
-                    result.add(String.format("Invalid output format."
-                            + " Found '%s' but expected '%s' or '%s'.",
-                            format, PLAIN_FORMAT_NAME, XML_FORMAT_NAME));
-                }
-            }
-            if (cmdLine.hasOption(OPTION_P_NAME)) {
-                final String propertiesLocation = cmdLine.getOptionValue(OPTION_P_NAME);
-                final File file = new File(propertiesLocation);
-                if (!file.exists()) {
-                    result.add(String.format("Could not find file '%s'.", propertiesLocation));
-                }
-            }
-            verifyThreadsNumberParameter(cmdLine, result, OPTION_CAPITAL_C_NAME,
-                "Checker threads number must be greater than zero",
-                "Invalid Checker threads number");
-            verifyThreadsNumberParameter(cmdLine, result, OPTION_CAPITAL_W_NAME,
-                "TreeWalker threads number must be greater than zero",
-                "Invalid TreeWalker threads number");
+        // return error if something is wrong in arguments
+        final List<File> filesToProcess = getFilesToProcess(options);
```

Comment:
getFilesToProcess need to be moved right after `execute`, to let read code as text by simply scrolling from to bottom.
the same problem is with getVersionString, just need to be moved to be right after `main`.



please do not be shocked, but we do really care about this :)
One day it will be completely automated, for now we did only half of way to this target.

Here is diagrams to help see it visually:
https://methods-distance.herokuapp.com/dsm?source_url=https://raw.githubusercontent.com/checkstyle/checkstyle/ea98b563e16b0c2bc09ee61d2824bdf0f1ea4a86/src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java
https://methods-distance.herokuapp.com/dsm?source_url=https://raw.githubusercontent.com/checkstyle/checkstyle/ea98b563e16b0c2bc09ee61d2824bdf0f1ea4a86/src/main/java/com/puppycrawl/tools/checkstyle/Main.java

first link - all is fine.
second link show a problem, minor problem.

Sorry that we are so fanatic about easy to read code.

Reply:
Done:

* moved `getFilesToProcess` to after `execute`
* moved `getVersionString` to after `main`